treescript/src/treescript/mercurial.py

```diff
@@ -347,7 +353,8 @@ def _count_outgoing(output):
         PushError: on failure
 
     """
-    source_repo = get_source_repo(task)
+    if not source_repo:
+        source_repo = get_source_repo(task)
     source_repo_ssh = source_repo.replace("https://", "ssh://")
     ssh_username = config.get("hg_ssh_user")
     ssh_key = config.get("hg_ssh_keyfile")
```

Comment:
The merge day instance only seemed to know about the first of those. Hmm. The existing treescript key is set up with puppet, right? Or since you mention scopes would you prefer those to be taskcluster secrets?

Reply:
I guess I misremembered the key names (and imagined the `tbird` one). The second one I was thinking of is `stage-ffxbld-merge` (see [here](https://github.com/mozilla-releng/build-puppet/blob/master/modules/mergeday/manifests/install.pp#L27)). They definitely should not be taskcluster secrets.

My suggestion about scopes was to have something like the `:bucket:*` scopes [here](https://github.com/mozilla-releng/scriptworker-scripts/blob/master/beetmoverscript/SCOPES.md) to control access to the different keys.

Reply:
Hm, they probably shouldn't be in the payload, if they're a scope, as the duplication won't add much. Is it likely there'll be someone who can use `stage-ffxbld-merge` and not the production one? 
We should really get these keys added to the usual place.

Reply:
> Hm, they probably shouldn't be in the payload, if they're a scope, as the duplication won't add much.

> Is it likely there'll be someone who can use `stage-ffxbld-merge` and not the production one?

Yes. My inclination would to make the key usable by anyone pushing to try:
- it is a level-1 key, so it doesn't grant any additional permissions.
- it make it easier for non-relengers to test changes (particularly to the in-tree config.